auth.test.js

```diff
@@ -0,0 +1,43 @@
+const jsc = require('jsverify');
+
+const auth = require('./auth'),
+      allCreds = require('./config').creds;
+      
+let token = false;
+
+describe('handler', () => {
+
+  test('should generate a token with a valid user', () => {
+    token = auth.handler(allCreds.kate);
+    expect(token).toBeTruthy();
+  });
+
+  // this is dumb and there's probably a better way not using property testing
+  test('should return false if the user is not valid', () => {
```

Comment:
This function is probably better tested with a static unit tests. I find that functions which operate on complex/large data or have many data transformations are more suited for property based testing. 

The heuristic I try to use is "When I write a property does it describe something non-obvious?". If so -> Property testing, otherwise -> unit tests.

The easiest example to explain is "inverse functions". If you have a pair of functions that are meant as the inverse of each other, a good property based test is "can I round trip for any data generated?". There are others, but I don't have any general guidance and am still learning.

Reply:
oh, so that's why you used the property tests on the generate and verify tests?